docs/articles/architecture/arch-vertical-configurations.rst

```diff
@@ -10,11 +10,110 @@ Each vertical configuration describes a sub-system of the product.
+   normal system and a secondary small partition can boot into a minimal
+   working system able to update the main partition. This solution is cheaper
+   but introduces a longer downtime than a symmetric partitioning.
+
+ - In-place upgrading where a single partition is capable of modifying itself in
+   an atomic way. This is notably used by OSTree. This requires less space than
+   the previous methods and a short downtime.
+
+If an update introduces regressions, the system should be able to revert
+automatically to a previous working state. This can be achieved using different
+mechanisms. Typical solutions include:
+
+ - With A/B symmetric partitioning, if one partition fails booting, the
+   bootloader can detect the error and boot the other partition.
+
+ - With a normal+recovery partitioning, if an error is detected, the bootloader
+   can reboot into recovery mode and fix the problem.
+
+ - With an in-place upgrading solution, specifically OSTree, different boot
+   entries can deploy different versions of the rootfs.
+
+The above requirement underlines the fact that an update mechanism can
+sometimes constrain the design of the rest of the system. An ideal update
+solution should make few assumptions on the rest of the system's architecture.
+(for instance: bootloader dependencies, number of partitions, read-only or
+read-write partitions etc...) This would limit the complexity of integration of
+the solution. Some update solutions also support several update mechanism
+schemes and give more flexibility to the platform developer.
+
+An update mechanism should limit the resources (i.e: Disk space, RAM usage,
+Bandwidth consumption etc...) usage on end-devices but also on the update
+servers. Typical solutions are:
+
+ - Differential updates: avoid too large downloads and processing. However,
+   this can cause problems if a local data block is corrupted.
+
+ - Full downloads: alternatively, downloading complete images solves the issue
+   of local corrupted data but is much more resource intensive.
+
+On a different side, updates can also be file or block based which affect the
+portability of update data across devices and also the size of downloaded
+information.
+
+An OTA server should propose various fleet and deployment management scenarios.
+For example, a car manufacturer should be able to deploy updates geographically
+or following a schedule.
+
+An update mechanism has to guarantee fundamental security capabilities. The two
+mainly expected features are:
+
+ - Integrity: this guarantees that data haven't been modified from the server
+   to the client. (for example, by man-in-the-middle attacks)
+
+ - Authentication: this guarantees that update have been created by an
+   authorized entity. (for example, by a Tier-1 or OEM vendor) This can be
+   achieved using digital signatures, CMAC or HMAC.
+
+In order to implement those checks correctly, it would be interesting to rely
+on a deep root of trust with TPM or TEE.
```

Comment:
Perhaps @e8johan can give some pointers on how this usually works?

Reply:
I'll be waiting for his comments, if any.